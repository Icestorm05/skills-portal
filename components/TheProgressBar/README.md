# The Progress Bar

## Functionality

The progress bar displays an indeterminate linear progress bar. This component should only be used once within the lifecycle of this application.

## Usage

### Props

`<the-progress-bar>` supports the following custom component attributes:

| attribute | type | description
| --- | --- | ---
| `color` | String *optional* | The color of the progress bar. Defaults to `secondary`. Accepts any of: `primary`, `accent`, `secondary`, `info`, `warning`, `error` or `success`.
| `height` | Number *optional* | The height of the progress bar. Defaults to `3`.

### Example

```html
<the-progress-bar v-if="display"
                  color="error"
                  :height="5"/>
```

```ts
import Component from 'nuxt-class-component';
import Vue from 'vue';

import TheProgressBar from '~/components/TheProgressBar/TheProgressBar.vue';

@Component({
    components: {
        TheProgressBar,
    },
})
export default class GenericComponent extends Vue {
    private display = false;

    // Show the progress bar, for 3 seconds.
    private mounted() {
        this.display = true;
        window.setTimeout(() => this.display = false, 3000);
    }
}
```