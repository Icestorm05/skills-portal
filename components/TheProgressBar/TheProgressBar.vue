<template lang="pug">
    v-progress-linear(indeterminate
                      :height="height"
                      :color="color").ma-0#TheProgressBar
</template>

<script lang="ts">
import Component from "nuxt-class-component";
import Vue from "vue";
import { Prop } from "vue-property-decorator";

import VProgressLinear from "vuetify/es5/components/VProgressLinear";

@Component({
  components: {
    VProgressLinear
  }
})
export default class TheProgressBar extends Vue {
  @Prop({ type: String, default: "secondary" })
  private color: string;
  @Prop({ type: Number, default: 4 })
  private height: number;
}
</script>
