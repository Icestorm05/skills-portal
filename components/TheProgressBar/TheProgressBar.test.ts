import TheProgressBar from "./TheProgressBar.vue";
import { mount } from "@vue/test-utils";

declare const expect: jest.Expect;

describe("The Progress Bar", () => {
  test("is a Vue instance", () => {
    const wrapper = mount(TheProgressBar);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test("snapshot has not changed", () => {
    const wrapper = mount(TheProgressBar);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("color prop changes the progress bar color", () => {
    const wrapper = mount(TheProgressBar, { propsData: { color: "error" } });
    expect(
      wrapper.find("#TheProgressBar .progress-linear__background").classes()
    ).toContain("error");
  });

  test("height prop changes the progress bar height", () => {
    const wrapper = mount(TheProgressBar, { propsData: { height: 5 } });
    expect(wrapper.find("#TheProgressBar").element.style.height).toEqual("5px");
  });
});
