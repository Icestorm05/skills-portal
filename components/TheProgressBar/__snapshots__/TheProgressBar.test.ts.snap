// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Progress Bar snapshot has not changed 1`] = `
<div class="progress-linear ma-0" style="height: 4px;" id="TheProgressBar">
    <div class="progress-linear__background secondary" style="height: 4px; opacity: 0.3; width: 100%;"></div>
    <div class="progress-linear__bar">
        <div class="progress-linear__bar__indeterminate progress-linear__bar__indeterminate--active">
            <div class="progress-linear__bar__indeterminate long secondary"></div>
            <div class="progress-linear__bar__indeterminate short secondary"></div>
        </div>
        <!---->
    </div>
</div>
`;
